### PR TITLE
Windows support

### LINKED ISSUE #6: Add support for building on Windows
Funny enough, it looks like my Windows build system had `bfd` and `libiberty` headers and library dependencies in `C:\Strawberry\c\include` and `C:\Strawberry\c\lib\` so these pre-requisites can be satisfied by installing [Strawberry Perl](https://strawberryperl.com/).

Developers can `choco install` the packages for `git cmake ninja doxygen.install graphviz` etc and get everything else needed.

When building, though, we need to set the `_WINDOWS` and `_WIN64` compile-time definitions, so that all the conditional compilation preprocessor statements in the Ghidra headers will work.

### PR DESCRIPTION
**Pull request by ekilmer**
~Not sure why building with Debug build type fails on Windows.~ Remove compiler warnings as errors

Not sure why tests are failing on Windows also (#92).

After further consideration, I plan to merge this PR even with failing Windows tests. I will open an issue to keep track of it (#92).

I've set some `Required` GitHub Actions tests to pass before merging PRs in the repo's settings, and this excludes the Windows builds, but we should try to remember to check the Windows runs to make sure that at least the build step passes.

closes #6

### CODE REVIEW

File: cmake/options.cmake [hunk 1]
```diff
@@ -22,6 +22,11 @@ option(sleigh_DFSVERIFY_DEBUG "Make sure that the block ordering algorithm produ
 option(sleigh_CPUI_STATISTICS "Turn on collection of cover and cast statistics")
 option(sleigh_CPUI_RULECOMPILE "Allow user defined dynamic rules")
 
+# Sanity checking
+if(CMAKE_HOST_SYSTEM_NAME STREQUAL "Windows")
+  set(sleigh_ENABLE_DOCUMENTATION OFF CACHE BOOL "Unsupported on Windows" FORCE)
```
Comment (tetsuo-cpp) [inline cmake/options.cmake:27]: Why is this?

**Review by tetsuo-cpp — APPROVED** [review]
LGTM aside from a few questions. By the way, do you have a Windows machine that you've been testing on or have you only been running on GH Actions?

File: tools/sleigh-lift/src/main.cpp [hunk 2]
```diff
@@ -261,7 +261,7 @@ int main(int argc, char *argv[]) {
   ContextInternal ctx;
   Sleigh engine(&load_image, &ctx);
   DocumentStorage storage;
-  Element *root = storage.openDocument(*sla_file_path)->getRoot();
+  Element *root = storage.openDocument(sla_file_path->string())->getRoot();
```
[the remark this answers is not in the document]
Comment (ekilmer) [reply tools/sleigh-lift/src/main.cpp:264]: Yeah, here is the error that I saw
```
166>D:\a\sleigh\sleigh\tools\sleigh-lift\src\main.cpp(264,54): error C2664: 'Document *DocumentStorage::openDocument(const std::string &)': cannot convert argument 1 from 'const _Ty' to 'const std::string &' [D:\a\sleigh\sleigh\build\tools\sleigh-lift\sleigh-lift.vcxproj]
                 with
                 [
                     _Ty=std::filesystem::path
                 ]
       D:\a\sleigh\sleigh\tools\sleigh-lift\src\main.cpp(264,54): message : Reason: cannot convert from 'const _Ty' to 'const std::string' [D:\a\sleigh\sleigh\build\tools\sleigh-lift\sleigh-lift.vcxproj]
                 with
                 [
                     _Ty=std::filesystem::path
                 ]
       D:\a\sleigh\sleigh\tools\sleigh-lift\src\main.cpp(264,40): message : No user-defined-conversion operator available that can perform this conversion, or the operator cannot be called [D:\a\sleigh\sleigh\build\tools\sleigh-lift\sleigh-lift.vcxproj]
       D:\a\sleigh\sleigh\build\include\sleigh\xml.hh(269,13): message : see declaration of 'DocumentStorage::openDocument' [D:\a\sleigh\sleigh\build\tools\sleigh-lift\sleigh-lift.vcxproj]
   166>D:\a\sleigh\sleigh\tools\sleigh-lift\src\main.cpp(288,64): error C2664: 'Document *DocumentStorage::openDocument(const std::string &)': cannot convert argument 1 from '_Ty' to 'const std::string &' [D:\a\sleigh\sleigh\build\tools\sleigh-lift\sleigh-lift.vcxproj]
                 with
                 [
                     _Ty=std::filesystem::path
                 ]
       D:\a\sleigh\sleigh\tools\sleigh-lift\src\main.cpp(288,64): message : Reason: cannot convert from '_Ty' to 'const std::string' [D:\a\sleigh\sleigh\build\tools\sleigh-lift\sleigh-lift.vcxproj]
                 with
                 [
                     _Ty=std::filesystem::path
                 ]
       D:\a\sleigh\sleigh\tools\sleigh-lift\src\main.cpp(288,48): message : No user-defined-conversion operator available that can perform this conversion, or the operator cannot be called [D:\a\sleigh\sleigh\build\tools\sleigh-lift\sleigh-lift.vcxproj]
       D:\a\sleigh\sleigh\build\include\sleigh\xml.hh(269,13): message : see declaration of 'DocumentStorage::openDocument' [D:\a\sleigh\sleigh\build\tools\sleigh-lift\sleigh-lift.vcxproj]
```
from this run (if the logs are still available) https://github.com/lifting-bits/sleigh/runs/6043352579?check_suite_focus=true#step:11:3364

**Comment by ekilmer** [comment]
> do you have a Windows machine that you've been testing on or have you only been running on GH Actions?

I use a VM for the most part, but I do have a personal Windows machine if I really need to dig into something.

**Comment by ekilmer** [comment]
Before I merge this, I want to take a stab at uploading the built Windows binaries only if tests fail

**Pull request merged**